plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/spreadsheet/SpreadsheetPresentation.java

```diff
@@ -2399,7 +2399,7 @@ private Color getCellBackground(
 
             if (cellSelected) {
                 Color normalColor = getCellBackground(attribute, row, cellValue, rowPosition, false, true);
-                if (normalColor == null || normalColor == backgroundNormal) {
+                if (normalColor == null || normalColor == backgroundNormal || UIStyles.isHighContrastTheme()) {
```

Comment:
This function is called thousands of times for big resultset rendering.
high contrast theme detection is a relatively slow function (it queries for thme service which crowls deeply in Eclipse API).
Lets cache this flag in the presention (along with other cache flags - they are refreshed in refreshData function)